scripts/shell.lua

```diff
@@ -305,6 +305,33 @@ function shell.read(replaceChar, size, history, colorFn)
         elseif k == "end" then
           strPos = #str + 1
           term.blink = 0
+        elseif k == "tab" and fileTabComplete then
+          local subStrPos = 1
+          local newStr = { }
+          for subStr in string.gmatch(str, "[^%s]+") do
```

Comment:
Pattern should be `%S+`

Reply:
Additionally, I tend to use the `str:func()` style across the project. (eg `str:gmatch("%S+")`)